src/Spice86.Core/Emulator/Memory/Memory.cs

```diff
@@ -253,30 +253,50 @@ public class Memory {
     }
 
     /// <summary>
-    ///     Allow a class to register for a certain memory range.
+    /// Allow a class to register for a certain memory range.
     /// </summary>
     /// <param name="baseAddress">The start of the frame</param>
     /// <param name="size">The size of the window</param>
     /// <param name="memoryDevice">The memory device to use</param>
     public void RegisterMapping(uint baseAddress, uint size, IMemoryDevice memoryDevice) {
         uint endAddress = baseAddress + size;
-        if (endAddress >= _memoryDevices.Length) {
-            Array.Resize(ref _memoryDevices, (int)endAddress);
+        if (endAddress >= _memoryDevices.Count) {
+            long length = endAddress - _memoryDevices.Count;
+            IMemoryDevice[] collection = new IMemoryDevice[length];
+            for (int i = 0; i < collection.Length; i++) {
+                collection[i] = memoryDevice;
+            }
+
+            _memoryDevices.AddRange(collection);
```

Comment:
Why would you need to support memory addresses above our highest supported memory address?